urllib3/poolmanager.py

```diff
@@ -129,22 +129,25 @@ def __exit__(self, exc_type, exc_val, exc_tb):
         # Return False to re-raise any potential exceptions
         return False
 
-    def _new_pool(self, scheme, host, port):
+    def _new_pool(self, scheme, host, port, pool_kwargs=None):
         """
-        Create a new :class:`ConnectionPool` based on host, port and scheme.
+        Create a new :class:`ConnectionPool` based on host, port, scheme, and
+        any additional pool keyword arguments.
 
-        This method is used to actually create the connection pools handed out
-        by :meth:`connection_from_url` and companion methods. It is intended
-        to be overridden for customization.
+        If `pool_kwargs` is provided, it overrides the instance's
+        `connection_pool_kw` variable. This method is used to actually create
```

Comment:
I don't think we should override the instance value, I think we should *merge* it. That is, if a key is specified in both dictionaries, the one from `pool_kwargs` should dominate the one from `self.connection_pool_kw`, but it should be possible to leave some unspecified.

This may require a custom method, because we'll probably want to allow *unsetting* some of these values by setting their value to `None`.